contracts/src/common/RegistryRolesMixin.sol

```diff
@@ -17,4 +17,7 @@ abstract contract RegistryRolesMixin {
 
     uint256 internal constant ROLE_SET_TOKEN_OBSERVER = 1 << 4;
     uint256 internal constant ROLE_SET_TOKEN_OBSERVER_ADMIN = ROLE_SET_TOKEN_OBSERVER << 128;
+
+    uint256 internal constant ROLE_UPGRADE = 1 << 5;
```

Comment:
If this role is only being used in one contract (`UserRegistry`) then it doesn't need to go into this file - it should stay within `UserRegistry` instead.

Reply:
this role could actually be available for any upgradable  logic as a future proof but I am alright moving it back to UserRegistry at this stage.